types/mirrorx/index.d.ts

```diff
@@ -0,0 +1,219 @@
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.4
+
+import * as React from 'react';
+import * as H from 'history';
+import {
+  ReactInstance, Component, ComponentState,
+  ReactElement, SFCElement, CElement,
+  DOMAttributes, DOMElement, ReactNode, ReactPortal
+} from 'react';
+import {
+  History,
+  Location,
+  Path,
+  LocationState,
+  LocationDescriptor
+} from 'history';
+
+import { Connect } from 'react-redux';
+import { match } from "react-router";
+
+
+export = mirror;
+export as namespace mirror;
+
+declare namespace mirror {
+
+  type Diff<T extends string, U extends string> = ({ [P in T]: P } & { [P in U]: never } & { [x: string]: never })[T]; 
+  type Omit<T, K extends keyof T> = Pick<T, Diff<keyof T, K>>;
+
+  interface model {
+      name:string,
+      initialState?:any,
+      reducers?:any,
+      effects?:any
+  } 
+
+  interface _model {
+    name:string,
+    reducers:any
+  }
+
+  interface LinkProps extends React.AnchorHTMLAttributes<HTMLAnchorElement> {
+    to: H.LocationDescriptor;
+    replace?: boolean;
+  }
+
+  interface NavLinkProps extends LinkProps {
+    activeClassName?: string;
+    activeStyle?: React.CSSProperties;
+    exact?: boolean;
+    strict?: boolean;
+    isActive?<P>(match: match<P>, location: H.Location): boolean;
+    location?: H.Location;
+  }
+
+  interface SwitchProps {
+    children?: React.ReactNode;
+    location?: H.Location;
+  }
+
+  interface RedirectProps {
+    to: H.LocationDescriptor;
+    push?: boolean;
+    from?: string;
+    path?: string;
+    exact?: boolean;
+    strict?: boolean;
+  }
+
+  interface RouteComponentProps<P> {
+    match: match<P>;
+    location: H.Location;
+    history: H.History;
+    staticContext?: any;
+  }
+
+  interface RouteProps {
+    location?: H.Location;
+    component?: React.ComponentType<RouteComponentProps<any>> | React.ComponentType<any>;
+    render?: ((props: RouteComponentProps<any>) => React.ReactNode);
+    children?: ((props: RouteComponentProps<any>) => React.ReactNode) | React.ReactNode;
+    path?: string;
+    exact?: boolean;
+    strict?: boolean;
+  }
+
+  interface PromptProps {
+    message: string | ((location: H.Location) => string | boolean);
+    when?: boolean;
+  }
+
+  interface LocationActionPayload {
+    method: string;
+    args?: any[];
+  }
+
+  interface RouterAction {
+    type:'@@router/CALL_HISTORY_METHOD',
+    payload: LocationActionPayload;
+  }
+  
+  interface routeActions {
+    push(location:any,state?:any):RouterAction,
+    replace(location:any,state?:any):RouterAction,
+    go(n:number):RouterAction,
+    goBack(): RouterAction,
+    goForward(): RouterAction
+  }
+
+  export interface ConnectedRouterProps<State> {
+    store?: any;
+    history?: History;
+}
+
+  interface Actions {
+    routing:routeActions,
+    [propName: string]: any
+  }
+
+  interface defaultOptions {
+    initialState?:{},
+    historyMode?:string,
+    middlewares:[any],
```

Comment:
Are you sure you don't want `any[]` here?

Reply:
Oh,mistake. sry, I will fix it.

Reply:
I have fixed it ,please review. Thanks.